src/components/PostDetails/PostDetails.jsx

```diff
@@ -1,45 +1,96 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+
+import PropTypes from 'prop-types';
+
+import { post, remove } from '../../api/api';
+import { getComments } from '../../api/comments';
+import { getPostDetails } from '../../api/posts';
+
 import { NewCommentForm } from '../NewCommentForm';
+
 import './PostDetails.scss';
 
-export const PostDetails = () => (
-  <div className="PostDetails">
-    <h2>Post details:</h2>
-
-    <section className="PostDetails__post">
-      <p>sunt aut facere repellat provident occaecati excepturi optio</p>
-    </section>
-
-    <section className="PostDetails__comments">
-      <button type="button" className="button">Hide 2 comments</button>
-
-      <ul className="PostDetails__list">
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>My first comment</p>
-        </li>
-
-        <li className="PostDetails__list-item">
-          <button
-            type="button"
-            className="PostDetails__remove-button button"
-          >
-            X
-          </button>
-          <p>sad sds dfsadf asdf asdf</p>
-        </li>
-      </ul>
-    </section>
-
-    <section>
-      <div className="PostDetails__form-wrapper">
-        <NewCommentForm />
-      </div>
-    </section>
-  </div>
-);
+export const PostDetails = ({ selectedPostId }) => {
+  const [postDetails, setPostDetails] = useState(0);
+  const [isVisibleComments, setVisibleComments] = useState(true);
+  const [comments, setComments] = useState([]);
+
+  useEffect(() => {
+    if (selectedPostId) {
+      getPostDetails(selectedPostId).then(setPostDetails);
+      getComments(selectedPostId).then(setComments);
+    } else {
+      setPostDetails(0);
+    }
+  }, [selectedPostId]);
+
+  const addComment = (name, email, body) => post('/comments', {
+    name, email, body, postId: selectedPostId,
+  }).then((newComment) => {
+    setComments(listOfComments => [newComment, ...listOfComments]);
```

Comment:
you need to make a new request after adding/removing a comment, not just changing your local state on the client